libs/partners/anthropic/langchain_anthropic/chat_models.py

```diff
@@ -837,6 +850,56 @@ def _lc_tool_calls_to_anthropic_tool_use_blocks(
     return blocks
 
 
+def _make_chat_generation_chunk(
+    text: str, message_dump: dict, full_generation_info: dict
+) -> Tuple[ChatGenerationChunk, dict]:
+    """Collect metadata and make ChatGenerationChunk.
+
+    Args:
+        text: text of the message chunk
+        message_dump: dict with metadata of the message chunk
+        full_generation_info: dict collecting metadata for full stream
+
+    Returns:
+        Tuple with ChatGenerationChunk and updated full_generation_info
+    """
+    generation_info = {}
+    usage_metadata: Optional[UsageMetadata] = None
+    for k, v in message_dump.items():
+        if k in ("content", "role", "type") or (
+            k in full_generation_info and k not in ("usage", "stop_reason")
+        ):
+            continue
+        elif k == "usage":
+            input_tokens = v.get("input_tokens", 0)
+            output_tokens = v.get("output_tokens", 0)
+            if "usage" not in full_generation_info:
+                full_generation_info[k] = v
+                usage_metadata = UsageMetadata(
+                    input_tokens=input_tokens,
+                    output_tokens=output_tokens,
+                    total_tokens=input_tokens + output_tokens,
+                )
+            else:
+                seen_input_tokens = full_generation_info[k].get("input_tokens", 0)
+                new_input_tokens = 0 if seen_input_tokens else input_tokens
```

Comment:
struggling to parse this line

Reply:
added a comment, lmk what you think.

Anthropic returns the input token count in every chunk. So without setting to zero, we'd have
```
{'input_tokens': 8, 'output_tokens': 1},
{'input_tokens': 8, 'output_tokens': 1},
{'input_tokens': 8, 'output_tokens': 1},
{'input_tokens': 8, 'output_tokens': 1},
```

We set to zero to avoid double-counting the input tokens (e.g., when summing chunks)
```
{'input_tokens': 8, 'output_tokens': 1},
{'input_tokens': 0, 'output_tokens': 1},
{'input_tokens': 0, 'output_tokens': 1},
{'input_tokens': 0, 'output_tokens': 1},
```
Just added an integration test to check for this behavior as well.

Reply:
oh oh makes sense